test/functional/mongo_client_options_tests.js

```diff
@@ -74,7 +74,7 @@ describe('MongoClient Options', function() {
   /**
    * @ignore
    */
-  it('should error on unexpected options', {
+  it.skip('should error on unexpected options', {
```

Comment:
🤔 does this mean our mapping doesn't really work? Why is this test skipped?

Reply:
Okay, I've fixed things now so the unsupported options message is consistent and options will only be validated on calls to `connect`. `validateOptions` can still be set at the constructor level (as it previously could be), but we will only warn about unknown options during `connect`. This maintains consistency with the existing behavior of the driver.